starknet_py/net/client.py

```diff
@@ -161,7 +161,8 @@ async def wait_for_tx(
                         if result.block_number is not None:
                             return result.block_number, status
                 elif status == TransactionStatus.REJECTED:
-                    raise TransactionRejectedError(result.rejection_reason)
+                    reason = result.rejection_reason or {}
+                    raise TransactionRejectedError(**reason)
```

Comment:
Same with changes to the `TransactionReceipt`, we'd prefer to keep the `TransactionRejectedError` oriented to RPC which doesn't include code. We could add a subclass of this exception specifically for the gateway_client, that incudes the code along with the message, so it doesn't confuse the RPC users.